### DISCUSSION TITLE
Expand option

### DISCUSSION THREAD

**Discussion by mbab** [DivinumOfficium/divinum-officium#3825] [Ideas]
Here is proposition to rework of 4 levels of expand option
0 = All - no changes here
1 = No common prayers (as _Pater_, _Deus in adjutorium_), but keep whole _Dómine Deus omnípotens_, _Respice_, etc.
2 = as above + psalm numbers and readings source or title for non biblical instead of full text
3 = Skeleton - no changes here

**Comment by APMarcello3** [comment]
I propose the term "lineamenta" in lieu of "skeleton".

**Comment by FAJ-Munich** [comment]
as of now, the Expansion of the `&psalm(50)` popup links do not work yet again. Not sure why.

For the Option 1: At the moment, the algorithm decided to not expand `$reference` starting with a Capital letter, keeping starting with miniscule (with the exception of `pater_noster`:  (cf. `horascommon.pl`line 2119:

```
 # Make popup link if we shouldn't expand.
    if ($expand =~ /none/i
      || ($expand !~ /all|skeleton/i && ($line =~ /^(?:[A-Z]|pater_noster)/)))
    {
      return setlink($sigil . $line, 0, $lang);
    }
```

I guess there could be a better logic to differentiate proper "Common Prayers"

**Comment by FAJ-Munich** [comment]
With the exception of implementing the "extraction of the readings titles" for `&lectio(1)` references, I have implemented the reworking you have proposed in this branch: https://github.com/FAJ-Munich/divinum-officium/tree/display-sep-626

The expansions are:

- **tota** : replacing the old _all_ (no changes)
- **psalteria** : no common prayers but the rest of the Psalterium is expanded (replacing the old _psalmi_)
- **propria** : only expanding the propers and all of the rubrics which are applying, e.g., to un-expanded common prayers
- **lineamenta** : replacing the _skeleton_ (no changes) as per Father's suggestion

@mbab  Could you, as time permits, double check if I missed out something? +LJCh+

**Reply by mbab** [reply]
@FAJ-Munich 
`-` sign in psalm numbering in `propria` looks strange
Eg. In 10-21-2024 Tridentine
![image](https://github.com/user-attachments/assets/3fc9aa97-da93-4344-9e3b-8f716c448b76)

also this (but have no ideo what to do with it)
![image](https://github.com/user-attachments/assets/949a19c7-1f1e-47e9-a711-8cb8c0d401e5)

There should by way to remove empty cells from below `lineamenta`
![image](https://github.com/user-attachments/assets/1f62525b-9064-4803-b1eb-7b2618eac8e6)

**Reply by FAJ-Munich** [reply]
Dziękuję bardzo za szybką recenzję!

> `-`sign ...
> `Psalm(231)

agreed. These need to be addressed alongside the meaningless `&lectio(1)` etc.; haven't had enough time to finish by yesterday evening.

> There should be a way to remove ...

I'll give it a try.

**Reply by FAJ-Munich** [reply]
Have rebased and force pushed my branch including the suggested implementations. 
If there are no objections I would open PR?

**Comment by APMarcello3** [comment]
![image](https://github.com/user-attachments/assets/5c603d4b-9320-4a74-a4ad-b25291a2f23c)

**Reply by FAJ-Munich** [reply]
fixed by #4124
